app/src/main/java/com/lucas/medicaltools/MedicalToolAdapter.kt

```diff
@@ -0,0 +1,93 @@
+package com.lucas.medicaltools
+
+import android.content.Context
+import android.graphics.Color
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.core.content.contentValuesOf
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.snackbar.Snackbar
+import com.lucas.medical_equip.repository.MedicalTool
+import com.squareup.picasso.Picasso
+import java.util.*
+import kotlin.collections.ArrayList
+
+class MedicalToolAdapter(private var medicalTools: List<MedicalTool>) : RecyclerView.Adapter<MedicalToolAdapter.ViewHolder>(), Filterable {
+    var medicalToolsFiltered = ArrayList<MedicalTool>()
+
+    init {
+        medicalToolsFiltered = medicalTools as ArrayList<MedicalTool>
+    }
+    inner class ViewHolder(medicalToolItemView: View) : RecyclerView.ViewHolder(medicalToolItemView) {
+        val medicalToolImage = itemView.findViewById<ImageView>(R.id.medical_tool_image)
+        val medicalToolDescription = itemView.findViewById<TextView>(R.id.medical_tool_text)
+        val isAvailable = itemView.findViewById<TextView>(R.id.medical_tool_avail)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val context = parent.context
+        val inflater = LayoutInflater.from(context)
+        //Inflate the view item
+        val medicalToolView = inflater.inflate(R.layout.medical_tool_item, parent, false)
+        return ViewHolder(medicalToolView)
+    }
+
+    override fun getItemCount(): Int {
+        return medicalToolsFiltered.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        //Get data model based on position
+        val medicalTool: MedicalTool = medicalToolsFiltered.get(position)
+
+        //Set the items views based on your views data model
+        val medicalToolImage = holder.medicalToolImage
+        val medicalToolDescription = holder.medicalToolDescription
+        val isMedicalAvail = holder.isAvailable
+        holder.itemView.setOnClickListener{onItemClicked(medicalTool, holder.medicalToolImage)}
```

Comment:
Lets pump up the complexity here on this item. 

Typically in Discovery, if we have an adapter where an item is tappable, we'd provide an observable for the Activity to decide what to do when tapping an item. It encourages re-usability of adapters.

So please make use of a Publish Subject which returns an observable for the activity to subscribe to and every time the user taps on an item, it's published via the Rx Java Publish Subject.